tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+After this scenario is implemented, I can install and configure Radius with my GitOps tool of choice. I can include Radius as a dependency for my Kubernetes cluster, just as I would any other dependency, via my GitOps tools. This ensures that I no longer have to manually manage the Radius instance I am providing for my developers and SREs to use, that there is an automated process to ensure that the deployed Radius instance matches the state specified I have specified in a repository. In other words, my experience with managing our Radius instance is consistent with how I normally use GitOps to define any other dependency in a Kubernetes environment.
+
+### Detailed Customer Experience
+ <!-- <List of steps the customer goes through from the start to the end of the scenario to provide more detailed view of exactly what the customer is able to do given the new capabilities>  -->
+<!-- Step 1
+Step 2
+… -->
+1. With my editor of choice (such as VSCode) open the GitOps (Flux or ArgoCD) Source File  
+1. Add installation of Radius as a Kubernetes cluster dependency for my desired Radius version.
+   1. Similar the params in the Flux schema defined here: [Helm Charts | Flux (fluxcd.io)](https://fluxcd.io/flux/components/source/helmcharts/)
+1. Commit the change to the git repo to submit my change and track it for future reference.
+1. GitOps tool will detect this repo change and update the relevant Kubernetes clusters 
+1. I can validate the Radius installation through various GitOps or Kubernetes tools. For example, I can run `kubectl get namespace` and verify Radius appears as Kubernetes namespace. 
+   1. I will be able to validate Radius on my Flux dashboard or my rad CLI/GitOps CLI as well once I've created a Radius Environment or have Radius Applications running.
+
+> As an operator, I'm happy with this experience because it is completely consistent with how I normally use Flux or ArgoCD to define any other dependency in a Kubernetes environment.
+ 
+The syntax for adding Radius as a Kubernetes cluster dependency must be:
+- Consistent with customer expectations for defining such a dependency 
+- Executable by Flux or ArgoCD (which based on current understanding should just work today). 
+
+> There is no new Radius specific user experience for this scenario. Customers are only interacting with Flux, ArgoCD, or their GitOps tool of choice.
+
+## Key investments
+<!-- List the features required to enable this scenario. -->
+Based on current understanding, there should be no new features that need to be implemented in Radius for this scenario. The key investment is in testing the GitOps tooling to ensure that the syntax for adding Radius as a Kubernetes cluster dependency is consistent with customer expectations for defining such a dependency and is executable by Flux or ArgoCD.
```

Comment:
The word we're looking for is Helm Chart. Radius is a Helm Chart, and gitops tools support that. 

I think you're right about the validation. Our chart needs to work.